p2p/peer.go

```diff
@@ -211,6 +224,10 @@ func (p *peer) OnStop() {
 	if err := p.mconn.Stop(); err != nil { // stop everything and close the conn
 		p.Logger.Debug("Error while stopping peer", "err", err)
 	}
+	for i := 0; i < len(p.channels); i++ {
+		p.recvListenersQuitChan <- struct{}{}
```

Comment:
Why do we need to write `len(channels)` messages? Isn't `close(chan)` won't result in all goroutines receiving a msg?